src/back/msl/mod.rs

```diff
@@ -298,10 +305,16 @@ impl Options {
 
     fn resolve_resource_binding(
         &self,
-        stage: crate::ShaderStage,
+        ep: &crate::EntryPoint,
         res_binding: &crate::ResourceBinding,
     ) -> Result<ResolvedBinding, EntryPointError> {
-        match self.per_stage_map[stage].resources.get(res_binding) {
+        let target = self
+            .per_entry_point_map
+            .as_ref()
+            .and_then(|map| map.get(&ep.name))
+            .and_then(|res| res.resources.get(res_binding))
+            .or_else(|| self.per_stage_map[ep.stage].resources.get(res_binding));
```

Comment:
Can we introduce a new fn on `Options` with the signature below and use it everywhere?
```rust
fn get_ep_resources<'ep>(&self, ep: &'ep EntryPoint) -> &PerStageResources
```

Reply:
Done. I changed `resolve_sizes_buffer` and `resolve_push_constants` to use the new helper. `resolve_resource_binding` is slightly different in that it falls back to `per_stage_map` only if the specific `res_binding` lookup fails (this allows the per stage map to define bindings that may not be present in the `per_entry_point_map`; which is why I left `resolve_resource_binding` as it is.

I could change the logic so that if `per_entry_point_map` is defined then `per_stage_map` gets ignored, which would be completely reasonable. I wanted to allow some flexibility but if `per_stage_map` will go away, then maybe allowing their combined use doesn't add much value.